File: frontend/lib/features/recipe/models/recipe_step.dart
//one numbered step in a recipe
//mirrors recipe_steps table
class RecipeStep {
  final int stepId;
  final int recipeId;
  final int stepNr;
  final String content;

  const RecipeStep({
    required this.stepId,
    required this.recipeId,
    required this.stepNr,
    required this.content,
  });

  factory RecipeStep.fromJson(Map<String, dynamic> json) {
    return RecipeStep(
      stepId: json['stepId'] as int,
      recipeId: json['recipeId'] as int,
      stepNr: json['stepNr'] as int,
      content: json['content'] as String,
    );
  }
  Map<String, dynamic> toJson() => {
        'stepNr': stepNr,
        'content': content,
      };
}

File: backend/src/main/java/com/mealchemy/recipe/model/RecipeStep.java
package com.mealchemy.recipe.model;

/* Import libraries */

import jakarta.persistence.*;

/* Import classes */

@Entity
@Table(name = recipe_steps)
public class RecipeStep
{
    /* Declaring fields */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "step_id")
    private int stepId;

    @Column(name = "recipe_id", nullable = false)
    private int recipeId;

    @Column(name = "step_nr", nullable = false)
    private int stepNr;

    @Column(name = "content", nullable = false)
    private String content;

    /* Getters */

    public int getStepId()
    {
        return stepId;
    }

    public int getRecipeId()
    {
        return recipeId;
    }

    public int getStepNr()
    {
        return stepNr;
    }

    public String getContent()
    {
        return content;
    }

    /* Setters */

    public void setStepNr(int stepNrIn)
    {
        stepNr = stepNrIn;
    }

    public void setContent(String contentIn)
    {
        content = contentIn;
    }
}